File: 1010.pairs-of-songs-with-total-durations-divisible-by-60/solution.cpp
class Solution {
public:
	int numPairsDivisibleBy60(vector<int>& time) {
		int cache[60] = {0};
		for (const int &i : time) {
			cache[i % 60]++;
		}
		int count = 0;
		for (const int &i : time) {
			cache[i % 60]--;
			count += cache[(60 - i % 60) % 60];
		}
		return count;
	}
};

File: 1010.pairs-of-songs-with-total-durations-divisible-by-60/solution.kt
class Solution {
    fun numPairsDivisibleBy60(time: IntArray): Int {
        val map = IntArray(60)
        var count = 0
        for (t in time) {
            count += map[(60 - t % 60) % 60] ?: 0
            map[t % 60]++
        }
        return count
    }
}
